pulpcore/pulpcore/app/models/repository.py

```diff
@@ -221,25 +207,147 @@ class RepositoryContent(Model):
 
         content (models.ForeignKey): The associated content.
         repository (models.ForeignKey): The associated repository.
+        vadded (models.ForeignKey): The RepositoryVersion which added the referenced Content.
+        vremoved (models.ForeignKey): The RepositoryVersion which removed the referenced Content.
     """
-    content = models.ForeignKey('Content', on_delete=models.CASCADE)
+    content = models.ForeignKey('Content', on_delete=models.CASCADE,
+                                related_name='version_memberships')
     repository = models.ForeignKey(Repository, on_delete=models.CASCADE)
+    vadded = models.ForeignKey('RepositoryVersion', related_name='added_memberships')
+    vremoved = models.ForeignKey('RepositoryVersion', null=True, related_name='removed_memberships')
+
+    class Meta:
+        unique_together = (('repository', 'content', 'vadded'),
+                           ('repository', 'content', 'vremoved'))
+
+
+class RepositoryVersion(Model):
+    """
+    A version of a repository's content set.
+    Fields:
+        number (models.PositiveIntegerField): A positive integer that uniquely identifies a version
+            of a specific repository. Each new version for a repo should have this field set to
+            1 + the most recent version.
+        created (models.DateTimeField): When the version was created.
+        action  (models.TextField): The action that produced the version.
+    Relations:
+        repository (models.ForeignKey): The associated repository.
+    """
+    SYNC = 'sync'
+    COPY = 'copy'
+    UPLOAD = 'upload'
+    ADD = 'add'
+    REMOVE = 'remove'
+    ACTIONS = (
+        (SYNC, 'Sync'),
+        (COPY, 'Copy'),
+        (UPLOAD, 'Upload'),
+        (ADD, 'Add'),
+        (REMOVE, 'Remove'),
+    )
+
+    repository = models.ForeignKey(Repository)
+    number = models.PositiveIntegerField(db_index=True, default=1)
+    created = models.DateTimeField(auto_now_add=True)
+    action = models.TextField(choices=ACTIONS)
```

Comment:
I don't see the value if tracking the `actions` and recommend removing.

Reply:
agreed, I'm +1 to removing it.